sdk/typescript/src/providers/json-rpc-provider.ts

```diff
@@ -795,4 +795,51 @@ export class JsonRpcProvider {
       EpochInfo,
     );
   }
+
+  /**
+   * Wait for a transaction block result to be available over the API.
+   * This can be used in conjunction with `executeTransactionBlock` to wait for the transaction to
+   * be available via the API.
+   * This currently polls the `getTransactionBlock` API to check for the transaction.
+   */
+  async waitForTransactionBlock({
+    signal,
+    timeout = 60 * 1000,
+    pollInterval = 2 * 1000,
+    ...input
+  }: {
+    /** An optional abort signal that can be used to cancel */
+    signal?: AbortSignal;
+    /** The amount of time to wait for a transaction block. Defaults to one minute. */
+    timeout?: number;
+    /** The amount of time to wait between checks for the transaction block. Defaults to 2 seconds. */
+    pollInterval?: number;
+  } & Parameters<
+    JsonRpcProvider['getTransactionBlock']
+  >[0]): Promise<SuiTransactionBlockResponse> {
+    const timeoutSignal = AbortSignal.timeout(timeout ?? 60 * 1000);
```

Comment:
huh cool, never new about `AbortSignal.timeout`

Reply:
Yeah it's a fun one! I wish abort signals were also then-able but I'll take what I can get